cmd/status.go

```diff
@@ -50,6 +50,74 @@ var statusUser string
 var statusName string
 var statusDetails bool
 var statusIncomplete bool
+var statusComplete bool
+var statusFailed bool
+var statusQueued bool
+
+type statusFilter struct {
+	incomplete bool
+	complete   bool
+	failed     bool
+	queued     bool
+}
+
+func newStatusFilter(incomplete, complete, failed, queued bool) *statusFilter {
+	set := false
+
+	for _, flag := range []bool{incomplete, complete, failed, queued} {
+		if flag {
+			if set {
+				die("--incomplete, --complete, --failed and --queued are mutually exclusive")
+			}
+
+			set = true
+		}
+	}
+
+	if !set {
+		return nil
+	}
+
+	return &statusFilter{
```

Comment:
This is fine, but the more idiomatic way would be to use an interface and a different concrete type for each different filter; that way you wouldn't need the switch statment on L108.

Reply:
Replaced with returning functions